ccl/importccl: TestImportUserDefinedTypes failed

ccl/importccl.TestImportUserDefinedTypes [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2900755&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2900755&tab=artifacts#/) on master @ [f451dd080c95fa73f7d796e1dc8bd6dd33eba09b](https://github.com/cockroachdb/cockroach/commits/f451dd080c95fa73f7d796e1dc8bd6dd33eba09b):


```
  github.com/cockroachdb/cockroach/pkg/sql/catalog/schemaexpr.ProcessColumnSet()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/schemaexpr/default_exprs.go:98 +0x33c
  github.com/cockroachdb/cockroach/pkg/sql/row.NewDatumRowConverter()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/row/row_converter.go:335 +0x664
  github.com/cockroachdb/cockroach/pkg/ccl/importccl.makeDatumConverter()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_base.go:449 +0x18c
  github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*parallelImporter).importWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_base.go:653 +0x7c
  github.com/cockroachdb/cockroach/pkg/ccl/importccl.runParallelImport.func1.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_base.go:553 +0xae
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GroupWorkers.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:175 +0x5d
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x4e
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x94

Goroutine 82 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x73
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:165 +0xe4
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GroupWorkers()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:175 +0xaa
  github.com/cockroachdb/cockroach/pkg/ccl/importccl.runParallelImport.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_base.go:552 +0x2f1
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x4e
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x94

Goroutine 137 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x73
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:165 +0xe4
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GroupWorkers()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:175 +0xaa
  github.com/cockroachdb/cockroach/pkg/ccl/importccl.runParallelImport.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_base.go:552 +0x2f1
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x4e
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x94
==================
E210420 05:17:51.591247 10975 jobs/adopt.go:260  [n1] 1  job 651511302561792001: adoption completed with error nodelocal://0/data173616058: process default and computed columns: type OID 100052 does not exist
    import_stmt_test.go:1317: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestImportUserDefinedTypes650710281
    testing.go:1038: race detected during execution of test
--- FAIL: TestImportUserDefinedTypes (5.32s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestImportUserDefinedTypes PKG=./pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
rohany marked as alumn{us/a}; resolving to solongordon instead
```
</p>
</details>

/cc @solongordon
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportUserDefinedTypes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/catalog/schemaexpr/default_exprs.go
+++ b/pkg/sql/catalog/schemaexpr/default_exprs.go
@@ -91,13 +91,15 @@ func ProcessColumnSet(
 
 	// Add all public or columns in DELETE_AND_WRITE_ONLY state
 	// that satisfy the condition.
+	ret := make([]catalog.Column, 0, len(tableDesc.AllColumns()))
+	ret = append(ret, cols...)
 	for _, col := range tableDesc.WritableColumns() {
 		if inSet(col) {
 			if !colIDSet.Contains(col.GetID()) {
 				colIDSet.Add(col.GetID())
-				cols = append(cols, col)
+				ret = append(ret, col)
 			}
 		}
 	}
-	return cols
+	return ret
 }

